test/06-batch.test.ts

```diff
@@ -0,0 +1,537 @@
+								studies_at__campus: 'bar',
+							},
+						},
+					],
+				})
+				.expect(
+					400,
+					'"GET and DELETE requests of a batch request must not have a body"',
+				);
+			await supertest(testLocalServer)
+				.post('/university/$batch')
+				.send({
+					requests: [
+						{
+							id: '0',
+							method: 'POST',
+							url: '/university/student',
+							body: {
+								matrix_number: 100003,
+								name: faker.name.firstName(),
+								last_name: faker.name.lastName(),
+								studies_at__campus: 'foo',
+							},
+						},
+						{
+							id: '1',
+							method: 'DELETE',
+							url: '/university/student',
+							body: {
+								matrix_number: 100004,
+								name: faker.name.firstName(),
+								last_name: faker.name.lastName(),
+								studies_at__campus: 'bar',
+							},
+						},
+					],
+				})
+				.expect(
+					400,
+					'"GET and DELETE requests of a batch request must not have a body"',
+				);
+		});
+
+		it('should fail if trying to query cross-model in one batch', async () => {
+			await supertest(testLocalServer)
+				.post('/university/$batch')
+				.send({
+					requests: [
+						{
+							id: '0',
+							method: 'POST',
+							url: '/v1/student',
+							body: {
+								matrix_number: 100003,
+								name: faker.name.firstName(),
+								last_name: faker.name.lastName(),
+								studies_at__campus: 'foo',
+							},
+						},
+						{
+							id: '1',
+							method: 'POST',
+							url: '/university/student',
+							body: {
+								matrix_number: 100004,
+								name: faker.name.firstName(),
+								last_name: faker.name.lastName(),
+								studies_at__campus: 'bar',
+							},
+						},
+					],
+				})
+				.expect(
+					400,
+					'"Batch requests must consist of requests for only one model"',
+				);
+		});
+
+		it('Should error if any authorization is passed in a request in the requests array', async () => {
+			await supertest(testLocalServer)
+				.post('/university/$batch')
+				.send({
+					requests: [
+						{
+							id: '0',
+							method: 'POST',
+							url: '/v1/student?apikey=some_key',
+							body: {
+								matrix_number: 100004,
+								name: faker.name.firstName(),
+								last_name: faker.name.lastName(),
+								studies_at__campus: 'foo',
+							},
+						},
+					],
+				})
+				.expect(
+					400,
+					'"Authorization may only be passed to the main batch request"',
+				);
+
+			await supertest(testLocalServer)
+				.post('/university/$batch')
+				.send({
+					requests: [
+						{
+							id: '0',
+							method: 'POST',
+							url: '/v1/student',
+							headers: { authorization: 'Bearer test' },
+							body: {
+								matrix_number: 100004,
+								name: faker.name.firstName(),
+								last_name: faker.name.lastName(),
+								studies_at__campus: 'foo',
+							},
+						},
+					],
+				})
+				.expect(
+					400,
+					'"Authorization may only be passed to the main batch request"',
+				);
+		});
+	});
+});
```

Comment:
I'm missing test cases for:

- request id request matches with returned response and Id. eg: mutliple gets for different resources, matching returned resources
- Expectation of two concurrent requests like 2 POST request sending the same matrix_number?
- At least some tests for $filter, $expand, ... Potentially a map of test parameters executed as test cases using the same structure [{testSet1}, {testSet2}, ...].foreach(it( run the test))

Reply:
- The 'successful request should have `responses` in its body' test sort of tests for the first bullet but I'll add a separate tests for multiple requests
- The requests should be sequential, so this shouldn't need testing, right?
- What exactly would this be for? This feature should not change how those work. Tests for $filter, $expand, etc. should live in their own test suite, no?